<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get cross references by gene symbol or name — get_xrefs_by_gene • ensemblr</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.9/font.css" rel="stylesheet"><link href="../deps/Inconsolata-0.4.9/font.css" rel="stylesheet"><link href="../deps/Source_Sans_Pro-0.4.9/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get cross references by gene symbol or name — get_xrefs_by_gene"><meta name="description" content="This function retrieves cross references by symbol or display name of a gene.
The data is returned as a tibble where each row is a
cross reference related to the provided symbol or display name of a gene. See
below under section Value for details about each column."><meta property="og:description" content="This function retrieves cross references by symbol or display name of a gene.
The data is returned as a tibble where each row is a
cross reference related to the provided symbol or display name of a gene. See
below under section Value for details about each column."><meta property="og:image" content="https://www.pattern.institute/ensemblr/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-none" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ensemblr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/api_coverage.html">Ensembl REST API Coverage</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/patterninstitute/ensemblr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Get cross references by gene symbol or name</h1>
      <small class="dont-index">Source: <a href="https://github.com/patterninstitute/ensemblr/blob/main/R/cross_references.R" class="external-link"><code>R/cross_references.R</code></a></small>
      <div class="d-none name"><code>get_xrefs_by_gene.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function retrieves cross references by symbol or display name of a gene.
The data is returned as a <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> where each row is a
cross reference related to the provided symbol or display name of a gene. See
below under section Value for details about each column.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_xrefs_by_gene</span><span class="op">(</span></span>
<span>  <span class="va">species_name</span>,</span>
<span>  <span class="va">gene</span>,</span>
<span>  ensembl_db <span class="op">=</span> <span class="st">"core"</span>,</span>
<span>  external_db <span class="op">=</span> <span class="st">""</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  warnings <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  progress_bar <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-species-name">species_name<a class="anchor" aria-label="anchor" href="#arg-species-name"></a></dt>
<dd><p>The species name, i.e., the scientific name, all letters
lowercase and space replaced by underscore. Examples: <code>'homo_sapiens'</code>
(human), <code>'ovis_aries'</code> (Domestic sheep) or <code>'capra_hircus'</code>
(Goat).</p></dd>


<dt id="arg-gene">gene<a class="anchor" aria-label="anchor" href="#arg-gene"></a></dt>
<dd><p>Symbol or display name of a gene, e.g., <code>'ACTB'</code> or
<code>'BRCA2'</code>.</p></dd>


<dt id="arg-ensembl-db">ensembl_db<a class="anchor" aria-label="anchor" href="#arg-ensembl-db"></a></dt>
<dd><p>Restrict the search to a database other than the default.
Ensembl's default database is <code>'core'</code>.</p></dd>


<dt id="arg-external-db">external_db<a class="anchor" aria-label="anchor" href="#arg-external-db"></a></dt>
<dd><p>Filter by external database, e.g. <code>'HGNC'</code>. An empty
string indicates no filtering.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Whether to be verbose about the http requests and respective
responses' status.</p></dd>


<dt id="arg-warnings">warnings<a class="anchor" aria-label="anchor" href="#arg-warnings"></a></dt>
<dd><p>Whether to show warnings.</p></dd>


<dt id="arg-progress-bar">progress_bar<a class="anchor" aria-label="anchor" href="#arg-progress-bar"></a></dt>
<dd><p>Whether to show a progress bar.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> of 12 variables:</p><dl><dt><code>species_name</code></dt>
<dd><p>Ensembl species name: this is the name used internally
by Ensembl to uniquely identify a species by name. It is the scientific
name but formatted without capitalisation and spacing converted with an
underscore, e.g., <code>'homo_sapiens'</code>.</p></dd>

<dt><code>gene</code></dt>
<dd><p>Gene symbol.</p></dd>

<dt><code>ensembl_db</code></dt>
<dd><p>Ensembl database.</p></dd>

<dt><code>primary_id</code></dt>
<dd><p>Primary identification in external database.</p></dd>

<dt><code>display_id</code></dt>
<dd><p>Display identification in external database.</p></dd>

<dt><code>external_db_name</code></dt>
<dd><p>External database name.</p></dd>

<dt><code>external_db_display_name</code></dt>
<dd><p>External database display name.</p></dd>

<dt><code>version</code></dt>
<dd><p>TODO</p></dd>

<dt><code>info_type</code></dt>
<dd><p>There are two types of external cross references (XRef):
direct (<code>'DIRECT'</code>) or dependent (<code>'DEPENDENT'</code>). A direct cross
reference is one that can be directly linked to a gene, transcript or
translation object in Ensembl Genomes by synonymy or sequence similarity. A
dependent cross reference is one that is transitively linked to the object
via the direct cross reference. The value can also be <code>'UNMAPPED'</code> for
unmapped cross references, or <code>'PROJECTION'</code> for TODO.</p></dd>

<dt><code>info_text</code></dt>
<dd><p>TODO</p></dd>

<dt>synonyms</dt>
<dd><p>Other names or acronyms used to refer to the
gene. Note that this column is of the list type.</p></dd>

<dt><code>description</code></dt>
<dd><p>Brief description of the external database entry.</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="ensembl-rest-api-endpoints">Ensembl REST API endpoints<a class="anchor" aria-label="anchor" href="#ensembl-rest-api-endpoints"></a></h2>
    <p><code>get_xrefs_by_gene()</code> makes GET requests to
<a href="https://rest.ensembl.org/documentation/info/xref_name" class="external-link">/xrefs/name/:species/:name</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Get cross references that relate to gene BRCA2</span></span></span>
<span class="r-in"><span><span class="fu">get_xrefs_by_gene</span><span class="op">(</span>species_name <span class="op">=</span> <span class="st">'human'</span>, gene <span class="op">=</span> <span class="st">'BRCA2'</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 9 × 12</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   species_name gene  ensembl_db primary_id display_id external_db_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> human        BRCA2 core       675        BRCA2      EntrezGene      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> human        BRCA2 core       1101       BRCA2      GeneCards       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> human        BRCA2 core       HGNC:1101  BRCA2      HGNC            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> human        BRCA2 core       A0A7P0TAP7 BRCA2      Uniprot_gn      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> human        BRCA2 core       A0A8V8TQQ4 BRCA2      Uniprot_gn      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> human        BRCA2 core       P51587     BRCA2      Uniprot_gn      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">7</span> human        BRCA2 core       A0A7P0T9D7 BRCA2      Uniprot_gn      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">8</span> human        BRCA2 core       675        BRCA2      WikiGene        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">9</span> human        BRCA2 core       IPR015525  BRCA2      Interpro        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 6 more variables: external_db_display_name &lt;chr&gt;, version &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   info_type &lt;chr&gt;, info_text &lt;chr&gt;, synonyms &lt;list&gt;, description &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ramiro Magno, Dany Mukesha, Isabel Duarte, Ana-Teresa Maia, CINTESIS, Pattern Institute.</p>
</div>

<div class="pkgdown-footer-right">

</div>

    </footer></div>





  </body></html>

