<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Ensembl REST API Coverage • ensemblr</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Ensembl REST API Coverage">
<meta property="og:description" content="">
<meta property="og:image" content="https://rmagno.eu/ensemblr/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ensemblr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/api_coverage.html">Ensembl REST API Coverage</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="api_coverage_files/kePrint-0.0.1/kePrint.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Ensembl REST API Coverage</h1>
            
      
      
      <div class="hidden name"><code>api_coverage.Rmd</code></div>

    </div>

    
    
<table class="table table table-striped table-hover table-condensed table-responsive">
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Endpoint
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ensemblr API
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Description
</th>
</tr></thead>
<tbody>
<tr grouplength="2">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Archive</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET archive/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Uses the given identifier to return its latest version
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST archive/id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieve the latest version for a set of identifiers
</td>
</tr>
<tr grouplength="12">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Comparative Genomics</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET cafe/genetree/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves a cafe tree of the gene tree using the gene tree stable identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET cafe/genetree/member/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves the cafe tree of the gene tree that contains the gene / transcript / translation stable identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET cafe/genetree/member/symbol/:species/:symbol
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves the cafe tree of the gene tree that contains the gene identified by a symbol
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET family/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves a family information using the family stable identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET family/member/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves the information for all the families that contains the gene / transcript / translation stable identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET family/member/symbol/:species/:symbol
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves the information for all the families that contains the gene identified by a symbol
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET genetree/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves a gene tree for a gene tree stable identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET genetree/member/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves the gene tree that contains the gene / transcript / translation stable identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET genetree/member/symbol/:species/:symbol
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves the gene tree that contains the gene identified by a symbol
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET alignment/region/:species/:region
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves genomic alignments as separate blocks based on a region and species
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET homology/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves homology information (orthologs) by Ensembl gene id
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET homology/symbol/:species/:symbol
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves homology information (orthologs) by symbol
</td>
</tr>
<tr grouplength="3">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Cross References</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET xrefs/symbol/:species/:symbol
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Looks up an external symbol and returns all Ensembl objects linked to it. This can be a display name for a gene/transcript/translation, a synonym or an externally linked reference. If a gene’s transcript is linked to the supplied symbol the service will return both gene and transcript (it supports transient links).
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET xrefs/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Perform lookups of Ensembl Identifiers and retrieve their external references in other databases
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET xrefs/name/:species/:name
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Performs a lookup based upon the primary accession or display label of an external reference and returning the information we hold about the entry
</td>
</tr>
<tr grouplength="3">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>EQTL</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET eqtl/stable_id/:species/:stable_id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns the p-value for each SNP in a given gene (e.g. ENSG00000227232)
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET eqtl/variant_name/:species/:variant_name
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns the p-values for a SNP (e.g. rs123)
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET eqtl/tissue/:species/
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns all tissues currently available in the DB
</td>
</tr>
<tr grouplength="26">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Information</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/analysis/:species
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
List the names of analyses involved in generating Ensembl data.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/assembly/:species
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
List the currently available assemblies for a species, along with toplevel sequences, chromosomes and cytogenetic bands.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/assembly/:species/:region_name
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns information about the specified toplevel sequence region for the given species.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/biotypes/:species
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
List the functional classifications of gene models that Ensembl associates with a particular species. Useful for restricting the type of genes/transcripts retrieved by other endpoints.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/biotypes/groups/:group/:object_type
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Without argument the list of available biotype groups is returned. With :group argument provided, list the properties of biotypes within that group. Object type (gene or transcript) can be provided for filtering.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/biotypes/name/:name/:object_type
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
List the properties of biotypes with a given name. Object type (gene or transcript) can be provided for filtering.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/compara/methods
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
List all compara analyses available (an analysis defines the type of comparative data).
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/compara/species_sets/:method
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
List all collections of species analysed with the specified compara method.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/comparas
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Lists all available comparative genomics databases and their data release. DEPRECATED: use info/genomes/division instead.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/data
</td>
<td style="text-align:left;">
<code><a href="../reference/get_data_versions.html">get_data_versions()</a></code>
</td>
<td style="text-align:left;width: 10cm; ">
Shows the data releases available on this REST server. May return more than one release (unfrequent non-standard Ensembl configuration).
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/eg_version
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns the Ensembl Genomes version of the databases backing this service
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/external_dbs/:species
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Lists all available external sources for a species.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/divisions
</td>
<td style="text-align:left;">
<code><a href="../reference/get_divisions.html">get_divisions()</a></code>
</td>
<td style="text-align:left;width: 10cm; ">
Get list of all Ensembl divisions for which information is available
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/genomes/:genome_name
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Find information about a given genome
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/genomes/accession/:accession
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Find information about genomes containing a specified INSDC accession
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/genomes/assembly/:assembly_id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Find information about a genome with a specified assembly
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/genomes/division/:division_name
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Find information about all genomes in a given division. May be large for Ensembl Bacteria.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/genomes/taxonomy/:taxon_name
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Find information about all genomes beneath a given node of the taxonomy
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/ping
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Checks if the service is alive.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/rest
</td>
<td style="text-align:left;">
<code><a href="../reference/get_rest_version.html">get_rest_version()</a></code>
</td>
<td style="text-align:left;width: 10cm; ">
Shows the current version of the Ensembl REST API.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/software
</td>
<td style="text-align:left;">
<code><a href="../reference/get_software_version.html">get_software_version()</a></code>
</td>
<td style="text-align:left;width: 10cm; ">
Shows the current version of the Ensembl API used by the REST server.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/species
</td>
<td style="text-align:left;">
<code><a href="../reference/get_species.html">get_species()</a></code>
</td>
<td style="text-align:left;width: 10cm; ">
Lists all available species, their aliases, available adaptor groups and data release.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/variation/:species
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
List the variation sources used in Ensembl for a species.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/variation/consequence_types
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Lists all variant consequence types.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/variation/populations/:species:/:population_name
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
List all individuals for a population from a species
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET info/variation/populations/:species
</td>
<td style="text-align:left;">
<code><a href="../reference/get_populations.html">get_populations()</a></code>
</td>
<td style="text-align:left;width: 10cm; ">
List all populations for a species
</td>
</tr>
<tr grouplength="3">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Linkage Disequilibrium</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ld/:species/:id/:population_name
</td>
<td style="text-align:left;">
<code><a href="../reference/get_ld_variants_by_window.html">get_ld_variants_by_window()</a></code>
</td>
<td style="text-align:left;width: 10cm; ">
Computes and returns LD values between the given variant and all other variants in a window centered around the given variant. The window size is set to 500 kb.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ld/:species/pairwise/:id1/:id2
</td>
<td style="text-align:left;">
<code><a href="../reference/get_ld_variants_by_window.html">get_ld_variants_by_pair()</a></code>
</td>
<td style="text-align:left;width: 10cm; ">
Computes and returns LD values between the given variants.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ld/:species/region/:region/:population_name
</td>
<td style="text-align:left;">
<code><a href="../reference/get_ld_variants_by_window.html">get_ld_variants_by_range()</a></code>
</td>
<td style="text-align:left;width: 10cm; ">
Computes and returns LD values between all pairs of variants in the defined region.
</td>
</tr>
<tr grouplength="4">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Lookup</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET lookup/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Find the species and database for a single identifier e.g. gene, transcript, protein
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST lookup/id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Find the species and database for several identifiers. IDs that are not found are returned with no data.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET lookup/symbol/:species/:symbol
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Find the species and database for a symbol in a linked external database
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST lookup/symbol/:species/:symbol
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Find the species and database for a set of symbols in a linked external database. Unknown symbols are omitted from the response.
</td>
</tr>
<tr grouplength="4">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Mapping</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET map/cdna/:id/:region
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Convert from cDNA coordinates to genomic coordinates. Output reflects forward orientation coordinates as returned from the Ensembl API.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET map/cds/:id/:region
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Convert from CDS coordinates to genomic coordinates. Output reflects forward orientation coordinates as returned from the Ensembl API.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET map/:species/:asm_one/:region/:asm_two
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Convert the co-ordinates of one assembly to another
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET map/translation/:id/:region
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Convert from protein (translation) coordinates to genomic coordinates. Output reflects forward orientation coordinates as returned from the Ensembl API.
</td>
</tr>
<tr grouplength="8">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Ontologies and Taxonomy</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ontology/ancestors/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Reconstruct the entire ancestry of a term from is_a and part_of relationships
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ontology/ancestors/chart/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Reconstruct the entire ancestry of a term from is_a and part_of relationships.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ontology/descendants/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Find all the terms descended from a given term. By default searches are conducted within the namespace of the given identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ontology/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Search for an ontological term by its namespaced identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ontology/name/:name
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Search for a list of ontological terms by their name
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET taxonomy/classification/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return the taxonomic classification of a taxon node
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET taxonomy/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Search for a taxonomic term by its identifier or name
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET taxonomy/name/:name
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Search for a taxonomic id by a non-scientific name
</td>
</tr>
<tr grouplength="3">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Overlap</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET overlap/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves features (e.g. genes, transcripts, variants and more) that overlap a region defined by the given identifier.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET overlap/region/:species/:region
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieves features (e.g. genes, transcripts, variants and more) that overlap a given region.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET overlap/translation/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Retrieve features related to a specific Translation as described by its stable ID (e.g. domains, variants).
</td>
</tr>
<tr grouplength="4">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Phenotype annotations</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET /phenotype/accession/:species/:accession
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return phenotype annotations for genomic features given a phenotype ontology accession
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET /phenotype/gene/:species/:gene
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return phenotype annotations for a given gene.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET /phenotype/region/:species/:region
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return phenotype annotations that overlap a given genomic region.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET /phenotype/term/:species/:term
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return phenotype annotations for genomic features given a phenotype ontology term
</td>
</tr>
<tr grouplength="7">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Regulation</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET regulatory/species/:species/microarray/:microarray/vendor/:vendor
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns information about a specific microarray
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET regulatory/species/:species/epigenome
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns information about all epigenomes available for the given species
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET species/:species/binding_matrix/:binding_matrix_stable_id/
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return the specified binding matrix
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET regulatory/species/:species/microarray
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns information about all microarrays available for the given species
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET regulatory/species/:species/microarray/:microarray/probe/:probe
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns information about a specific probe from a microarray
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET regulatory/species/:species/microarray/:microarray/probe_set/:probe_set
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns information about a specific probe_set from a microarray
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET regulatory/species/:species/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns a RegulatoryFeature given its stable ID (e.g. ENSR00000006733)
</td>
</tr>
<tr grouplength="4">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Sequence</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET sequence/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Request multiple types of sequence by stable identifier. Supports feature masking and expand options.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST sequence/id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Request multiple types of sequence by a stable identifier list.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET sequence/region/:species/:region
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Returns the genomic sequence of the specified region of the given species. Supports feature masking and expand options.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST sequence/region/:species
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Request multiple types of sequence by a list of regions.
</td>
</tr>
<tr grouplength="1">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Transcript Haplotypes</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET transcript_haplotypes/:species/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Computes observed transcript haplotype sequences based on phased genotype data
</td>
</tr>
<tr grouplength="6">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Variation</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET variant_recoder/:species/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Translate a variant identifier, HGVS notation or genomic SPDI notation to all possible variant IDs, HGVS and genomic SPDI
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST variant_recoder/:species
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Translate a list of variant identifiers, HGVS notations or genomic SPDI notations to all possible variant IDs, HGVS and genomic SPDI
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET variation/:species/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Uses a variant identifier (e.g. rsID) to return the variation features including optional genotype, phenotype and population data
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET variation/:species/pmcid/:pmcid
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Fetch variants by publication using PubMed Central reference number (PMCID)
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET variation/:species/pmid/:pmid
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Fetch variants by publication using PubMed reference number (PMID)
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST variation/:species/
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Uses a list of variant identifiers (e.g. rsID) to return the variation features including optional genotype, phenotype and population data
</td>
</tr>
<tr grouplength="22">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>Variation GA4GH</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ga4gh/beacon
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return Beacon information
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ga4gh/beacon/query
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return the Beacon response for allele information
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST ga4gh/beacon/query
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return the Beacon response for allele information
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ga4gh/features/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return the GA4GH record for a specific sequence feature given its identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST ga4gh/features/search
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return a list of sequence annotation features in GA4GH format
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST ga4gh/callsets/search
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return a list of sets of genotype calls for specific samples in GA4GH format
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ga4gh/callsets/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return the GA4GH record for a specific CallSet given its identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST ga4gh/datasets/search
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return a list of datasets in GA4GH format
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ga4gh/datasets/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return the GA4GH record for a specific dataset given its identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST ga4gh/featuresets/search
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return a list of feature sets in GA4GH format
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ga4gh/featuresets/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return the GA4GH record for a specific featureSet given its identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ga4gh/variants/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return the GA4GH record for a specific variant given its identifier.
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST ga4gh/variantannotations/search
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return variant annotation information in GA4GH format for a region on a reference sequence
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST ga4gh/variants/search
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return variant call information in GA4GH format for a region on a reference sequence
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST ga4gh/variantsets/search
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return a list of variant sets in GA4GH format
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ga4gh/variantsets/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return the GA4GH record for a specific VariantSet given its identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST ga4gh/references/search
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return a list of reference sequences in GA4GH format
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ga4gh/references/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return data for a specific reference in GA4GH format by id
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST ga4gh/referencesets/search
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return a list of reference sets in GA4GH format
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ga4gh/referencesets/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return data for a specific reference set in GA4GH format
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST ga4gh/variantannotationsets/search
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return a list of annotation sets in GA4GH format
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET ga4gh/variantannotationsets/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Return meta data for a specific annotation set in GA4GH format
</td>
</tr>
<tr grouplength="6">
<td colspan="3" style="border-bottom: 1px solid;">
<strong>VEP</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET vep/:species/hgvs/:hgvs_notation
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Fetch variant consequences based on a HGVS notation
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST vep/:species/hgvs
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Fetch variant consequences for multiple HGVS notations
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET vep/:species/id/:id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Fetch variant consequences based on a variant identifier
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST vep/:species/id
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Fetch variant consequences for multiple ids
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
GET vep/:species/region/:region/:allele/
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Fetch variant consequences
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
POST vep/:species/region
</td>
<td style="text-align:left;">
TODO
</td>
<td style="text-align:left;width: 10cm; ">
Fetch variant consequences for multiple regions
</td>
</tr>
</tbody>
</table>
</div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Ramiro Magno, Ana-Teresa Maia, Centre for Biomedical Research, University of Algarve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
